File: day20_part1_2015.py
with open("input.txt", "r") as file:
    puzzle_input = int(file.read().strip())

def factors_sum(n):
    total = 0
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            if n / i == i:
                total += i
            else:
                total += i + n/i
    return total

house_num = 1
while factors_sum(house_num) * 10 < puzzle_input:
    house_num += 1

print(house_num)

File: training_data/reflection-tuning/day20_part1_2015.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Read input from file
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	scanner.Scan()
	target, _ := strconv.Atoi(scanner.Text())

	// Find the lowest house number
	house := 1
	for {
		presents := calculatePresents(house)
		if presents >= target {
			fmt.Println(house)
			break
		}
		house++
	}
}

func calculatePresents(house int) int {
	sum := 0
	for i := 1; i*i <= house; i++ {
		if house%i == 0 {
			sum += i * 10
			if i*i != house {
				sum += (house / i) * 10
			}
		}
	}
	return sum
}
